pkg/security/resolvers/cgroup/model/model.go

```diff
@@ -59,52 +58,54 @@ type CacheEntry struct {
 	sync.RWMutex
 	Deleted          *atomic.Bool
 	WorkloadSelector WorkloadSelector
-	PIDs             *simplelru.LRU[uint32, int8]
+	PIDs             map[uint32]int8
 }
 
 // NewCacheEntry returns a new instance of a CacheEntry
 func NewCacheEntry(id string, pids ...uint32) (*CacheEntry, error) {
-	pidsLRU, err := simplelru.NewLRU[uint32, int8](1000, nil)
-	if err != nil {
-		return nil, err
-	}
-
 	newCGroup := CacheEntry{
 		Deleted: atomic.NewBool(false),
 		ContainerContext: model.ContainerContext{
 			ID: id,
 		},
-		PIDs: pidsLRU,
+		PIDs: make(map[uint32]int8, 10),
 	}
 
 	for _, pid := range pids {
-		newCGroup.PIDs.Add(pid, 0)
+		newCGroup.PIDs[pid] = 1
 	}
 	return &newCGroup, nil
 }
 
-// GetPIDs returns the list of root pids for the current workload
+// GetPIDs returns the list of pids for the current workload
 func (cgce *CacheEntry) GetPIDs() []uint32 {
 	cgce.RLock()
 	defer cgce.RUnlock()
 
-	return cgce.PIDs.Keys()
+	pids := make([]uint32, len(cgce.PIDs))
+	i := 0
+	for k := range cgce.PIDs {
```

Comment:
for i, k := range cge.PIDs {
    pids[i] =k
}

Reply:
In the loop you suggested, the `i` variable is the actual pid we want to add to the slice and not an index.